### ISSUE TITLE
Invite team / organization members (b2b)

### ISSUE THREAD

**Issue by mffap** [zitadel/zitadel#7083]
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

In a standard b2b scenario I want to either invite team members by sending them a link to their email or allow them to sign-up via SSO. Currently ZITADEL doesn't offer this functionality, but only a variant that might not suit the need for b2b scenarios.

### Describe your ideal solution

As Manager, I can invite users to join the organization, so I can onboard my team / users.

Considerations: 

- Users can be invited by email
- The email contains a link to register for the organization from which the invite was initiated
- If SSO is enabled, I can instead sign-up with the external IDP
- Email texts can be customized
- Managers that invite team members can assign a role to the user before sending the invite (eg, engineer)
- Managers don't have to provide a password when creating an invitation
- The validity time for the invitation should be limited

### Environment

ZITADEL Cloud

### Additional Context

Related issues / discussions: 
- https://github.com/zitadel/zitadel/issues/7026
- https://github.com/zitadel/zitadel/issues/7029
- https://github.com/zitadel/zitadel/discussions/4771
- https://github.com/zitadel/zitadel/discussions/5563
- https://github.com/zitadel/zitadel/issues/8310
- https://github.com/zitadel/typescript/issues/166

**Comment by mffap** [comment]
@hifabienne I've opened this issue based on our discussion around strengthening the b2b use cases. When you sign-up to a new cloud service, the first user ("Admin") should have the option to easily invite their team members to join the organization. Although we have ways around it, we should make this scenario possible.

**Comment by hifabienne** [comment]
I added the issue to the road map and for the moment to the backlog.
As it requires the #6433 to be finished first, otherwise it is not possible to create users with only email addresses within ZITADEL. Without that we will not be able to fully implement this process within ZITADEL.

**Comment by sschoeb** [comment]
I had a discussion with Fabienne in Discord about a specific "sub-feature" of this feature. So I thought I add it here so it may not get lost. 

An invitation will need something that the user can be authenticated (e.g. a code). This is useful if you want to show additional information on the registrations creen for the user (e.g. his email, or the name of the organisation he is about to join). This code has to be reusable as it is a use case that a user does open the invitation but does not finish it. Therefore if he opens it a second time it should still be valid.

**Comment by shreeharsha-factly** [comment]
If you can provide invitation APIs it will be helpful for Inviter and Invitee to track invitations within their application.

Existing Users:
If an existing user is invited to join another organization, they should be able to view the invitation list and accept or decline the invitation from their application.

Invitation Tracking:
Invitees should be able to track the status of their invitations, whether they have accepted or declined. In the case of an expired invitation, they can resend the invitation.

**Comment by muhlemmer** [comment]
@hifabienne as I see that all the linked issues are closed. Can we close this epic?

**Comment by hifabienne** [comment]
@muhlemmer yes technical implementation is done, but as it is currently only fully available with the new login ui, and only partially with the old. I need to check how and where we want to document that process

**Comment by hifabienne** [comment]
The invite flow is implemented with the session/user API v2.
For the full feature set, you will have to use the Zitadel v3 in combination with our new Login V2.
